sdk/resourcemanager/Azure.ResourceManager/src/ArmClient.cs

```diff
@@ -114,12 +113,6 @@ private void CopyApiVersionOverrides(ArmClientOptions options)
             foreach (var keyValuePair in options.ResourceApiVersionOverrides)
             {
                 ApiVersionOverrides.Add(keyValuePair.Key, keyValuePair.Value);
-                if (!ResourceApiVersionCache.TryGetValue(keyValuePair.Key.Namespace, out var apiVersionCache))
-                {
-                    apiVersionCache = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
-                    ResourceApiVersionCache.TryAdd(keyValuePair.Key.Namespace, apiVersionCache);
```

Comment:
What's the impact if `ResourceApiVersionCache` is not updated here?

Reply:
More details can be found in the PR description, instead of updating `ResourceApiVersionCache` here which is before fetching from remote, we should update it after fetching from remote.